flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/utils/JarHandlerUtils.java

```diff
@@ -26,13 +36,42 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import static org.apache.flink.runtime.rest.handler.util.HandlerRequestUtils.fromRequestBodyOrQueryParameter;
+import static org.apache.flink.runtime.rest.handler.util.HandlerRequestUtils.getQueryParameter;
+import static org.apache.flink.shaded.guava18.com.google.common.base.Strings.emptyToNull;
+
 /**
  * Utils for jar handlers.
  *
  * @see org.apache.flink.runtime.webmonitor.handlers.JarRunHandler
  * @see org.apache.flink.runtime.webmonitor.handlers.JarPlanHandler
  */
 public class JarHandlerUtils {
+	/** Parse program arguments in jar run or plan request. */
+	public static <R extends JarRequestBody, M extends MessageParameters> List<String> getProgramArgs(
+		HandlerRequest<R , M> request, Logger log) throws RestHandlerException {
+		JarRequestBody requestBody = request.getRequestBody();
+		List<String> programArgs = tokenizeArguments(
+			fromRequestBodyOrQueryParameter(
+				emptyToNull(requestBody.getProgramArguments()),
+				() -> getQueryParameter(request, ProgramArgsQueryParameter.class),
+				null,
+				log));
+		List<String> programArgsList = fromRequestBodyOrQueryParameter(
+			requestBody.getProgramArgumentsList(),
+			() -> request.getQueryParameter(ProgramArgsListQueryParameter.class),
+			null,
+			log);
+		if (requestBody.getProgramArgumentsList() != null) {
+			if (!programArgs.isEmpty()) {
+				throw new IllegalArgumentException(
```

Comment:
neither handler is handling this exception, so the user will get an INTERNAL_SERVER_ERROR instead of BAD_REQUEST.